packages/gitbook/src/components/Search/server-actions.tsx

```diff
@@ -245,69 +198,72 @@ export async function streamRecommendedQuestions(args: { siteSpaceId?: string })
 /**
  * Search for content in a site by scoping the search to all content, a specific spaces or current space.
  */
-async function searchSiteContent(
-    context: GitBookBaseContext,
-    args: {
-        query: string;
-        scope:
-            | { mode: 'all' }
-            | { mode: 'current'; siteSpaceId: string }
-            | { mode: 'specific'; siteSpaceIds: string[] };
-    }
-): Promise<OrderedComputedResult[]> {
-    const { dataFetcher } = context;
-    const siteURLData = await getSiteURLDataFromMiddleware();
-
-    const { scope, query } = args;
-
-    if (query.length <= 1) {
-        return [];
-    }
-
-    const [searchResults, { structure }] = await Promise.all([
-        throwIfDataError(
-            dataFetcher.searchSiteContent({
-                organizationId: siteURLData.organization,
-                siteId: siteURLData.site,
-                query,
-                scope,
-            })
-        ),
-        throwIfDataError(
-            dataFetcher.getPublishedContentSite({
-                organizationId: siteURLData.organization,
-                siteId: siteURLData.site,
-                siteShareKey: siteURLData.shareKey,
-            })
-        ),
-    ]);
-
-    return (
-        await Promise.all(
-            searchResults.map((spaceItem) => {
-                const found = findSiteSpaceBy(
-                    structure,
-                    (siteSpace) => siteSpace.space.id === spaceItem.id
-                );
-                const siteSection = found?.siteSection;
-                const siteSectionGroup = found?.siteSectionGroup;
-
-                return Promise.all(
-                    spaceItem.pages.map((pageItem) =>
-                        transformSitePageResult(context, {
-                            pageItem,
-                            spaceItem,
-                            siteSpace: found?.siteSpace,
-                            space: found?.siteSpace.space,
-                            spaceURL: found?.siteSpace.urls.published,
-                            siteSection: siteSection ?? undefined,
-                            siteSectionGroup: (siteSectionGroup as SiteSectionGroup) ?? undefined,
-                        })
-                    )
-                );
-            })
-        )
-    ).flat(2);
+export async function searchSiteContent({
+    query,
+    ...scope
+}: {
+    query: string;
+} & (
+    | { mode: 'all' }
+    | { mode: 'current'; siteSpaceId: string }
+    | { mode: 'specific'; siteSpaceIds: string[] }
+)): Promise<OrderedComputedResult[]> {
+    return traceErrorOnly(`Search.searchSiteContent.${scope.mode}`, async () => {
+        if (query.length <= 1) {
+            return [];
+        }
+
+        const [context, { organization, site, shareKey }] = await Promise.all([
+            getServerActionBaseContext(),
+            getSiteURLDataFromMiddleware(),
+        ]);
```

Comment:
The refactoring in this file allows us to parallelize the `getServerActionBaseContext` and `getSiteURLDataFromMiddleware` calls.